### PR TITLE
Fetch Twitter

### PR DESCRIPTION
**Pull request by rock3r**
## Problem

We want to move the social feed fetching to the backend

## Solution

Implement fetching the feed. TODO: cron this to regularly fetch updates

### Test(s) added

Nope

### Paired with 

@fourlastor

### CODE REVIEW

File: functions/src/twitter/twitter-data.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+export interface Tweet {
+    id_str: string
+    full_text: string
+    created_at: string
+    display_text_range: [number, number]
+    user: User
+    entities: Entities
+    retweeted_status: any
+    in_reply_to_screen_name: string | null
```
Comment (fourlastor) [inline functions/src/twitter/twitter-data.ts]: This should be using optional

File: functions/src/twitter/fetch-twitter.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        // Exclude retweets
+        return rawTweet.retweeted_status === undefined || rawTweet.retweeted_status === null
+    })
+        .map((rawTweet): FirestoreTweet => ({
+            id: rawTweet.id_str,
+            text: rawTweet.full_text,
+            createdAt: new Date(rawTweet.created_at),
+            displayTextRange: rawTweet.display_text_range,
+            user: firestoreUserFrom(rawTweet.user),
+            inReplyToScreenName: rawTweet.in_reply_to_screen_name,
```
Comment (fourlastor) [inline functions/src/twitter/fetch-twitter.ts]: Here we should be transforming possible undefined values to null (undefined would probably break insertion in firestore)

File: functions/src/twitter/fetch-twitter.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        })
+}
+
+const uploadToFirestore = (firestore: Firestore, tweets: Tweet[]): Promise<WriteResult>[] => {
+    const tweetsCollection = firestore.collection('social_stream')
+        .doc('twitter')
+        .collection('tweets')
+
+    return tweets.filter((rawTweet) => {
+        // Exclude retweets
```
Comment (fourlastor) [inline functions/src/twitter/fetch-twitter.ts]: Worth extracting a named function and remove the comment?

File: functions/src/twitter/fetch-twitter.ts (hunk 2 above)
Comment (rock3r) [reply functions/src/twitter/fetch-twitter.ts]: Seems to work so far but sure, better safe than sorry. It's done with `|| null` right?

File: functions/src/twitter/fetch-twitter.ts (hunk 3 above)
Comment (rock3r) [reply functions/src/twitter/fetch-twitter.ts]: Yup will do

File: functions/src/twitter/twitter-data.ts (hunk 1 above)
Comment (rock3r) [reply functions/src/twitter/twitter-data.ts]: You created this :P will change 👍

File: functions/src/twitter/fetch-twitter.ts (hunk 2 above)
Comment (fourlastor) [reply functions/src/twitter/fetch-twitter.ts]: Yeah that should do

Comment (rock3r) [reply functions/src/twitter/fetch-twitter.ts]: 7a87754f895d81b866659034ff55bfa2e62d4220

File: functions/src/twitter/twitter-data.ts (hunk 1 above)
Comment (rock3r) [reply functions/src/twitter/twitter-data.ts]: 7a87754f895d81b866659034ff55bfa2e62d4220

File: functions/src/twitter/fetch-twitter.ts (hunk 3 above)
Comment (rock3r) [reply functions/src/twitter/fetch-twitter.ts]: d612b397f3bae512709063db993b75cf3a379d12

**Pull request merged**